index.html

```diff
@@ -1987,6 +2132,432 @@ <h2>
         </ol>
       </section>
     </section>
+    <section data-cite="file-system-access">
+      <h2>
+        File Handler Items
+      </h2>
+      <p>
+        Each <dfn data-local-lt="file handler">file handler</dfn> represents a
+        URL in the scope of the application that can handle launches with
+        [=file types=] it accepts. It has the following members:
+      </p>
+      <ul>
+        <li>[=file handler/action=]
+        </li>
+        <li>[=file handler/name=]
+        </li>
+        <li>[=file handler/accept=]
+        </li>
+        <li>[=file handler/icons=]
+        </li>
+      </ul>
+      <p>
+        A user agent can use these members to associate the web application
+        with [=file type=] on the operating system.
+      </p>
+      <p>
+        A <dfn>file type</dfn> can be defined by a [=mime type=] and/or [=file
+        extension=]. A file belongs to a file type if the OS determines it to
+        have a mime type and/or its name ends with a certain [=file
+        extension=]. A <dfn>file extension</dfn> is a string that start with a
+        "." and only contains [=valid suffix code points=]. A <dfn>valid suffix
+        code point</dfn> is a [=code point=] that is [=ASCII alphanumeric=],
+        U+002B (+), or U+002E (.). Additionally, [=file extensions=] are are
+        limited to a length of 16 code points.
+      </p>
+      <aside class="note">
+        <p>
+          Note: These code points were chosen to support most pre-existing file
+          formats. The vast majority of file extensions are purely
+          alphanumeric, but compound extensions (such as .tar.gz) and
+          extensions such as .c++ for C++ source code are also fairly common,
+          hence the inclusion of + and . as allowed code points.
+        </p>
+      </aside>
+      <aside class="note">
+        <p>
+          The selection of an app to handle a file launch if there are multiple
+          apps registered for the file's [=file type=] is functionality left to
+          the operating system.
+        </p>
+      </aside>
+      <section>
+        <h3>
+          `action` member
+        </h3>
+        <p>
+          The [=file handler=]'s <code><dfn data-dfn-for=
+          "file handler">action</dfn></code> member is a <a>string</a> that
+          represents a relative URL of the [=manifest/start_url=] origin that
+          is [=manifest/within scope=] of a [=Document/processed manifest=].
+          This URL will be navigated to in the steps to [=execute a file
+          handler launch=].
+        </p>
+      </section>
+      <section>
+        <h3>
+          `name` member
+        </h3>
+        <p>
+          The [=file handler=]'s <code><dfn data-dfn-for=
+          "file handler">name</dfn></code> member is a <a>string</a> that
+          describes the file type. User agents MAY pass this information to the
```

Comment:
Something doesn't seem right here (might just be the wording)... it's a "name" that "describes". 

Also, I'm a bit unsure where this gets shown to a user? Is there potential abuse here? Like `"name": "Pictures"` and `"type": "application/whatever"`.

Reply:
> Something doesn't seem right here (might just be the wording)... it's a "name" that "describes".

Being descriptive is a property of a good name, I think :) But I shall change this to "identifies".

> Also, I'm a bit unsure where this gets shown to a user?

The non-normative note right below describes where it's shown to the user. Do you have suggestions on how to make that text more clear?

> Is there potential abuse here? Like "name": "Pictures" and "type": "application/whatever"

I can't see a security threat here, but then it's not my specialty. We have, however, subjected this API to several rounds of internal security review and while spoofing in general is a keen concern, the name is not as much of a focus as the icon. I left the wording here more relaxed ("MAY") to accommodate that, depending on the level of trust the user agent places in the app. For example, a user agent with a lot of ceremony around installation may trust installed apps enough to display the name and icon, whereas a user agent where installation is totally frictionless may not want to display name and icon at all.